src/bootstrap/compile.rs

```diff
@@ -501,6 +501,20 @@ impl Step for Rustc {
         let mut cargo = builder.cargo(compiler, Mode::Rustc, SourceType::InTree, target, "build");
         rustc_cargo(builder, &mut cargo, target);
 
+        if let Some(path) = &builder.config.rust_profile_generate {
+            if compiler.stage == 1 {
+                cargo.rustflag(&format!("-Cprofile-generate={}", path));
+                // Apparently necessary to avoid overflowing the counters during
+                // a Cargo build profile
+                cargo.rustflag("-Cllvm-args=-vp-counters-per-site=4");
+            }
+        }
+        if let Some(path) = &builder.config.rust_profile_use {
+            if compiler.stage == 1 {
+                cargo.rustflag(&format!("-Cprofile-use={}", path));
+            }
+        }
+
```

Comment:
We'll probably also need the following LLVM flags (in both the `-generate` and the `-use` case) so that we don't end up with absolute file system paths in profile data:

https://github.com/michaelwoerister/rust/commit/d5413a6576e5d8eaacaa6c0f36c2a5e68871b363#diff-ca6526d91d41c31fad645c46417e0f1ce61e324542994e8c8b72b1f7a08c91bdR898-R899